src/components/DeleteConfirmation.js

```diff
@@ -0,0 +1,26 @@
+import Popup from "./Popup.js";
+
+export default class DeleteConfirmation extends Popup {
+  constructor({ popupSelector, handleDeleteSubmit }) {
+    super({ popupSelector });
+    this._popupButton = this._popupElement.querySelector(".modal__button");
+    this._handleDeleteSubmit = handleDeleteSubmit;
+    this._card = null;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._popupButton.addEventListener("click", (evt) => {
+      evt.preventDefault();
+      this._handleDeleteSubmit(this._card, this._id);
+      this.close();
```

Comment:
` this.close();`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**

Reply:
My remark hasn't been corrected https://github.com/Tonyg8642/se_project_aroundtheus/pull/1#discussion_r2036208121 


` this.close();` should be deleted from line 16 inside `DeleteConfirmation`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**